### PR TITLE
Add docs for gctuner in pd

### PR DESCRIPTION
**Pull request by hnes**
Signed-off-by: Sen Han <00hnes@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

issue: tikv/pd#5827

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@hnes Please invite one or two tech reviewers. Thanks~

**Comment by easonn7** [comment]
LGTM

File: pd-configuration-file.md [hunk 1]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
+### `server-memory-limit`
+
++ PD 实例的内存限制。
++ 默认值： `0.8`
++ 最小值： `0.01`
++ 最大值： `0.99`
+
+### `server-memory-limit-gc-trigger`
+
++ PD 尝试触发 GC 的阈值。当 PD 的内存使用达到 `server-memory-limit` 值 * `server-memory-limit-gc-trigger` 值时，则会主动触发一次 Golang GC。在一分钟之内只会主动触发一次 GC。
```
Comment (shichun-0415) [inline pd-configuration-file.md:116]:
```suggestion
+ PD 尝试触发 GC 的阈值。当 PD 的内存使用达到 `server-memory-limit` 值 * `server-memory-limit-gc-trigger` 值时，则会主动触发一次 Golang GC (GOGC)。在一分钟之内只会主动触发一次 GC。
```

File: pd-configuration-file.md [hunk 2]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
++ 默认值： `0.7`
++ 最小值： `0.5`
++ 最大值： `0.99`
```
Comment (shichun-0415) [inline pd-configuration-file.md:119]:
```suggestion
+ 最大值： `0.99`
+ 单位：%
```

File: pd-configuration-file.md [hunk 3]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
 
 pd-server 相关配置项。
 
+### `server-memory-limit`
+
++ PD 实例的内存限制。
++ 默认值： `0.8`
++ 最小值： `0.01`
++ 最大值： `0.99`
```
Comment (shichun-0415) [inline pd-configuration-file.md:112]:
```suggestion
+ 最大值：`0.99`
+ 单位：%
```

File: pd-configuration-file.md [hunk 4]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
+
+### `server-memory-limit-gc-trigger`
+
++ PD 尝试触发 GC 的阈值。当 PD 的内存使用达到 `server-memory-limit` 值 * `server-memory-limit-gc-trigger` 值时，则会主动触发一次 Golang GC。在一分钟之内只会主动触发一次 GC。
++ 默认值： `0.7`
++ 最小值： `0.5`
```
Comment (shichun-0415) [inline pd-configuration-file.md:118]:
```suggestion
+ 默认值：`0.7`
+ 最小值：`0.5`
```

File: pd-configuration-file.md [hunk 5]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
 
 pd-server 相关配置项。
 
+### `server-memory-limit`
+
++ PD 实例的内存限制。
++ 默认值： `0.8`
++ 最小值： `0.01`
```
Comment (shichun-0415) [inline pd-configuration-file.md:111]:
```suggestion
+ 默认值：`0.8`
+ 最小值：`0.01`
```

File: pd-configuration-file.md [hunk 6]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
+### `server-memory-limit-gc-trigger`
+
++ PD 尝试触发 GC 的阈值。当 PD 的内存使用达到 `server-memory-limit` 值 * `server-memory-limit-gc-trigger` 值时，则会主动触发一次 Golang GC。在一分钟之内只会主动触发一次 GC。
++ 默认值： `0.7`
++ 最小值： `0.5`
++ 最大值： `0.99`
+
+### `enable-gogc-tuner`
+
++ 是否开启 GOGC Tuner。
```
Comment (shichun-0415) [inline pd-configuration-file.md:123]:
```suggestion
+ 是否开启 GOGC Tuner 来调节内存使用。开启该功能后，XXX，缓解 GC 压力。
```

Comment (shichun-0415) [reply pd-configuration-file.md:123]: 是否可以加上开启该功能后的效果？

File: pd-configuration-file.md [hunk 7]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
++ 默认值： `true`
+
+### `gc-tuner-threshold`
+
++ GOGC Tuner 自动调节的最大内存阈值，即 `server-memory-limit` 值 * `server-memory-limit-gc-trigger` 值，超过阈值后 GOGC Tuner 会停止工作。
++ 默认值： `0.6`
++ 最小值： `0`
```
Comment (shichun-0415) [inline pd-configuration-file.md:130]:
```suggestion
+ 默认值：`0.6`
+ 最小值：`0`
```

File: pd-configuration-file.md [hunk 8]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
++ 最大值： `0.9`
```
Comment (shichun-0415) [inline pd-configuration-file.md:131]:
```suggestion
+ 最大值：`0.9`
+ 单位：%
```

File: pd-configuration-file.md (hunk 6 above)
Comment (hnes) [reply pd-configuration-file.md:123]: 也许并不需要在这里写？参考 TiDB 文档中关于 [tidb_enable_gogc_tuner 的用法描述](https://docs.pingcap.com/tidb/stable/system-variables#tidb_enable_gogc_tuner-new-in-v640)也很简单

File: pd-configuration-file.md (hunk 1 above)
Comment (hnes) [reply pd-configuration-file.md:116]: 应该并不需要加这个注解，Golang GC 应该就足够了

File: pd-configuration-file.md (hunk 3 above)
Comment (hnes) [reply pd-configuration-file.md]: Thanks a lot.

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: pd-configuration-file.md [hunk 9]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
 
+### `server-memory-limit` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
++ PD 实例的内存限制比例。`0` 值表示不设内存限制。
++ 默认值：`0`
++ 最小值：`0`
++ 最大值：`0.99`
+
+### `server-memory-limit-gc-trigger` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
```
Comment (TomShawn) [inline pd-configuration-file.md:115]:
```suggestion

> **警告：**
>
> 在当前版本中，该配置项为实验特性，不建议在生产环境中配置该项。

```

File: pd-configuration-file.md [hunk 10]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
++ PD 尝试触发 GC 的阈值比例。当 PD 的内存使用达到 `server-memory-limit` 值 * `server-memory-limit-gc-trigger` 值时，则会主动触发一次 Golang GC。在一分钟之内只会主动触发一次 GC。
++ 默认值：`0.7`
++ 最小值：`0.5`
++ 最大值：`0.99`
+
+### `enable-gogc-tuner` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
```
Comment (TomShawn) [inline pd-configuration-file.md:122]:
```suggestion

> **警告：**
>
> 在当前版本中，该配置项为实验特性，不建议在生产环境中配置该项。

```

File: pd-configuration-file.md [hunk 11]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
++ 是否开启 GOGC Tuner。
++ 默认值：`false`
+
+### `gc-tuner-threshold` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
```
Comment (TomShawn) [inline pd-configuration-file.md:127]:
```suggestion

> **警告：**
>
> 在当前版本中，该配置项为实验特性，不建议在生产环境中配置该项。

```

File: pd-configuration-file.md (hunk 9 above)
Comment (shichun-0415) [reply pd-configuration-file.md:119]: 建议：在当前版本中，PD 实例的内存限制为实验特性，不建议在生产环境中修改该配置项。

File: pd-configuration-file.md (hunk 10 above)
Comment (shichun-0415) [reply pd-configuration-file.md:130]: 开启 GOGC Tuner 似乎是在 6.4 就引入的功能，见系统变量[`tidb_enable_gogc_tuner`](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_enable_gogc_tuner-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5)

File: pd-configuration-file.md (hunk 11 above)
Comment (shichun-0415) [reply pd-configuration-file.md:139]: 建议：在当前版本中，PD 实例的内存限制为实验特性，不建议在生产环境中修改该配置项。

File: pd-configuration-file.md [hunk 12]
```diff
@@ -104,6 +104,32 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
 
 pd-server 相关配置项。
 
+### `server-memory-limit` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
```
[the remark this answers is not in the document]
Comment (shichun-0415) [reply pd-configuration-file.md:108]: 不建议在生产环境中修改该配置项的默认值？

File: pd-configuration-file.md (hunk 10 above)
Comment (hnes) [reply pd-configuration-file.md:130]: [tidb_enable_gogc_tuner](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_enable_gogc_tuner-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) 是 tidb-server 中的配置，当前 pr 改的是 pd 的配置

File: pd-configuration-file.md [hunk 13]
```diff
@@ -104,6 +104,48 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
++ 最小值：`0`
++ 最大值：`0.99`
+
+### `server-memory-limit-gc-trigger` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> 在当前版本中，该配置项为实验特性，不建议在生产环境中使用。
+
++ PD 尝试触发 GC 的阈值比例。当 PD 的内存使用达到 `server-memory-limit` 值 * `server-memory-limit-gc-trigger` 值时，则会主动触发一次 Golang GC。在一分钟之内只会主动触发一次 GC。
```
Comment (Oreoxmt) [inline pd-configuration-file.md:124]:
```suggestion
+ PD 尝试触发 GC 的阈值比例。当 PD 的内存使用达到 `server-memory-limit * server-memory-limit-gc-trigger` 时，则会主动触发一次 Golang GC。在一分钟之内只会主动触发一次 GC。
```

File: pd-configuration-file.md [hunk 14]
```diff
@@ -104,6 +104,48 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
++ 是否开启 GOGC Tuner。
++ 默认值：`false`
+
+### `gc-tuner-threshold` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> 在当前版本中，该配置项为实验特性，不建议在生产环境中使用。
+
++ GOGC Tuner 自动调节的最大内存阈值比例，即 `server-memory-limit` 值 * `server-memory-limit-gc-trigger` 值，超过阈值后 GOGC Tuner 会停止工作。
```
Comment (Oreoxmt) [inline pd-configuration-file.md:144]:
```suggestion
+ GOGC Tuner 自动调节的最大内存阈值比例，即 `server-memory-limit * server-memory-limit-gc-trigger`，超过阈值后 GOGC Tuner 会停止工作。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: pd-configuration-file.md (hunk 13 above)
Comment (hnes) [reply pd-configuration-file.md:124]: Thanks a lot for your comments. The phrasing in these series parameters' doc is mainly referred from TiDB's doc,  please see [here](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_server_memory_limit_gc_trigger-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5). So I think maybe the current description is also acceptable.

File: pd-configuration-file.md (hunk 14 above)
Comment (hnes) [reply pd-configuration-file.md:144]: Please see [here](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_server_memory_limit_gc_trigger-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5).

**Pull request merged**